js/classes/entities/midground/enemies/eel.js

```diff
@@ -0,0 +1,235 @@
+     * 
+     * @returns {string} "left" or "right"
+     */
+    getFacing() {
+        return (this.getDirection() < 0) ? "left" : "right";
+    }
+
+    /**
+     * Get the direction in which the enemy is currently facing.
+     * 
+     * @returns {number} -1 for left and 1 for right
+     */
+    getDirection() {
+        return this.targetX - this.getCenter().x > 0 ? 1 : -1;
+    }
+
+    /**
+     * Get the center of the enemy.
+     * @returns {Vector} the center of the enemy
+     */
+    getCenter() {
+        return Vector.add(this.pos, new Vector(Eel.SCALED_SIZE.x / 2, Eel.SCALED_SIZE.y / 2));
+    }
+    
+    /** Change what the Eel is doing and where it is. */
+    update() {
+        this.targetX = CHAD.getCenter().x;
+        const deathAnim = this.animations[this.getFacing()]["dying"];
+
+        if (this.health > 0) {
+            const secondsSinceLastAttack = Date.now() / 1000 - this.lastAttack;
+
+            // if we've finished our current attack, change action to idle
+            if (this.action === "attacking" 
+                && this.animations[this.getFacing()]["attacking"].totalTime < secondsSinceLastAttack) {
+                    
+                this.action = "idle";
+            }
+    
+            // if Chad is close enough, bite him
+            if (this.chadDistance() < Eel.SCALED_SIZE.x / 2) {
+                if (secondsSinceLastAttack > Eel.ATTACK_COOLDOWN) {
+                    console.log("attacking")
+                    this.action = "attacking";
+                    // if it's been long enough, start a new attack 
+                    this.animations[this.getFacing()]["attacking"].elapsedTime = 0;
+                    this.lastAttack = Date.now() / 1000;
+                    this.dealtDamage = false;
+                } else if (this.action === "attacking" 
+                    && secondsSinceLastAttack > Eel.DAMAGE_DELAY && !this.dealtDamage) {
+                    // if we're at the proper point in our attack animation, deal damage
+    
+                    CHAD.takeDamage(Eel.ATTACK_DAMAGE);
+                    this.dealtDamage = true;
+                }
+            }
+        }
+
+        if (this.currentJumpHeight <= 0) {
+            this.isJumping = true;
+        } 
+
+        if (this.currentJumpHeight >= this.maxJumpHeight) {
+            this.isJumping = false;
+        }
+
+        // console.log("currentJumpHeight: " + this.currentJumpHeight, "maxJumpHeight: " + this.maxJumpHeight)
+
+        if (this.isJumping) {
+            // jump up
+            this.pos = Vector.add(this.pos, new Vector(0, -this.speed * GAME.clockTick));
+            this.currentJumpHeight += this.speed * GAME.clockTick;
+        } 
+        else {
+            // fall down
+            this.pos = Vector.add(this.pos, new Vector(0, this.speed * GAME.clockTick));
+            this.currentJumpHeight -= this.speed * GAME.clockTick;
+        }
```

Comment:
If you want, you can probably achieve a more natural jumping effect by applying a one-time force, similar to how we do it elsewhere, when the eel reaches a certain y-position.

Reply:
What do you mean? Where else do we apply one-time forces?

Reply:
Enemies jump by setting their y-velocity to a specific jump velocity at the very beginning of the jump, and I believe Chad does something similar as well.